examples/with_promises/recurring.js

```diff
@@ -1,52 +1,173 @@
+    referenceId: exampleReferenceId,
+    customerId: exampleCustomerId,
+    recurringAction: 'PAYMENT',
+    currency: 'IDR',
+    amount: 1000,
+    paymentMethods: [
+      {
+        paymentMethodId: examplePaymentMethodId,
+        rank: 1,
+      },
+    ],
+    scheduleId: scheduleId,
+    immediateActionType: null,
+    notification_config: {
+      recurring_created: ['SMS'],
+      recurring_succeeded: ['SMS'],
+      recurring_failed: ['SMS'],
+    },
+    failedCycleAction: 'STOP',
+    metadata: null,
   })
-  .then(({ id }) => rp.stopPayment({ id }))
-  .then(r => {
-    // eslint-disable-next-line no-console
-    console.log('recurring payment stopped:', r.id);
-    return r;
+    .then(createdPlan => {
+      planId = createdPlan.id;
+      console.log('created recurring plan:', createdPlan);
+      return createdPlan;
+    })
+    .then(createdPlan => {
+      return r.getPlan({
+        id: createdPlan.id,
+        businessId: exampleBusinessId,
+      });
+    })
+    .then(plan => {
+      console.log('recurring plan:', plan);
+      return plan;
+    })
+    .then(plan => {
+      return r.editPlan({
+        id: plan.id,
+        businessId: exampleBusinessId,
+        amount: 1000,
+      });
+    })
+    .then(editedPlan => {
+      console.log('edited recurring plan:', editedPlan);
+      return editedPlan;
+    })
+    .then(plan => {
+      return r.deactivatePlan({
+        id: plan.id,
+        businessId: exampleBusinessId,
+      });
+    })
+    .then(deactivatedPlan => {
+      console.log('deactivated recurring plan:', deactivatedPlan);
+    });
+}
+
+function runCycles() {
+  r.getAllCycles({
+    planId: planId,
+    businessId: exampleBusinessId,
+    limit: 2,
   })
-  .catch(e => {
-    console.error(e); // eslint-disable-line no-console
-    process.exit(1);
-  });
+    .then(() => {
+      console.log('recurring cycles:', response.data);
+      console.log('has more recurring cycles:', response.has_more);
+    })
+    .then(() => {
+      return r.getCycle({
+        id: response.data[0].id,
+        planId: planId,
+        businessId: exampleBusinessId,
+      });
+    })
+    .then(cycle => {
+      console.log('recurring cycle:', cycle);
+      return cycle;
+    })
+    .then(cycle => {
+      return r.editCycle({
+        id: cycle.id,
+        planId: planId,
+        businessId: exampleBusinessId,
+        amount: 1000,
+        currency: 'IDR',
+        scheduledTimestamp: scheduledTimestamp,
+      });
+    })
+    .then(editedCycle => {
+      console.log('edited recurring cycle:', editedCycle);
+      return editedCycle;
+    })
+    .then(editedCycle => {
+      return r.cancelCycle({
+        id: editedCycle.id,
+        planId: planId,
+        businessId: exampleBusinessId,
+      });
+    })
+    .then(canceledCycle => {
+      console.log('canceled recurring cycle:', canceledCycle);
+    });
+}
+
+(function() {
+  runSchedules()
+    .then(() => runPlans())
```

Comment:
![image](https://user-images.githubusercontent.com/22883661/159434199-1da0e0ec-fdc3-41eb-bddc-64b3cb9776e2.png)
Error here

Reply:
fixed, thank you